src/DataTypes/Serializations/SerializationInterval.cpp

```diff
@@ -0,0 +1,273 @@
+
+#include "SerializationCustomSimpleText.h"
+
+#include <Columns/IColumn.h>
+#include <DataTypes/DataTypeFactory.h>
+#include <DataTypes/DataTypeInterval.h>
+#include <Formats/FormatSettings.h>
+#include <IO/WriteBuffer.h>
+#include <Parsers/Kusto/ParserKQLTimespan.h>
+
+#include <magic_enum.hpp>
+
+#include <concepts>
+#include <span>
+
+namespace DB
+{
+using ColumnInterval = DataTypeInterval::ColumnType;
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+    extern const int NOT_IMPLEMENTED;
+    extern const int VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE;
+}
+}
+
+namespace
+{
+class SerializationKQLInterval : public DB::SerializationCustomSimpleText
+{
+public:
+    explicit SerializationKQLInterval(DB::IntervalKind kind_) : SerializationCustomSimpleText(nullptr), kind(kind_) { }
+
+    void serializeText(const DB::IColumn & column, size_t row, DB::WriteBuffer & ostr, const DB::FormatSettings & settings) const override;
+    void deserializeText(DB::IColumn & column, DB::ReadBuffer & istr, const DB::FormatSettings & settings, bool whole) const override;
+
+private:
+    DB::IntervalKind kind;
+};
+
+void SerializationKQLInterval::serializeText(
+    const DB::IColumn & column, const size_t row, DB::WriteBuffer & ostr, const DB::FormatSettings &) const
+{
+    const auto * interval_column = checkAndGetColumn<DB::ColumnInterval>(column);
+    if (!interval_column)
+        throw DB::Exception(DB::ErrorCodes::ILLEGAL_COLUMN, "Expected column of underlying type of Interval");
+
+    const auto & value = interval_column->getData()[row];
+    const auto ticks = kind.toAvgNanoseconds() * value / 100;
+    const auto interval_as_string = DB::ParserKQLTimespan::compose(ticks);
+    ostr.write(interval_as_string.c_str(), interval_as_string.length());
+}
+
+void SerializationKQLInterval::deserializeText(
+    [[maybe_unused]] DB::IColumn & column,
+    [[maybe_unused]] DB::ReadBuffer & istr,
+    [[maybe_unused]] const DB::FormatSettings & settings,
+    [[maybe_unused]] const bool whole) const
+{
+    throw DB::Exception(
+        DB::ErrorCodes::NOT_IMPLEMENTED, "Deserialization is not implemented for {}", kind.toNameOfFunctionToIntervalDataType());
+}
+
+template <typename... Args, std::invocable<const DB::ISerialization *, Args...> Method>
+void dispatch(
+    std::span<const std::unique_ptr<const DB::ISerialization>> serializations,
+    const Method method,
+    const DB::FormatSettings::IntervalFormat format,
+    Args &&... args)
+{
+    const auto format_index = magic_enum::enum_index(format);
+    if (!format_index)
+        throw DB::Exception(DB::ErrorCodes::VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE, "No such format exists");
+
+    const auto & serialization = serializations[*format_index];
+    if (!serialization)
+        throw DB::Exception(DB::ErrorCodes::NOT_IMPLEMENTED, "Option {} is not implemented", magic_enum::enum_name(format));
+
+    (serialization.get()->*method)(std::forward<Args>(args)...);
```

Comment:
I'd rather use `if`, avoid the `serializations` array altogether, and remove the separate ParserKQLTimespan by serializing in-place here; it will also allow avoiding extra string copy.

Reply:
I'm okay with going with a conditional expression, so no worries there.

As for the `ParserKQLTimespan` class, I extracted this from our implementation of KQL that is yet to be contributed, so there's some significance to it being separate (https://github.com/ClibMouse/ClickHouse/blob/Kusto-phase3/src/Parsers/Kusto/ParserKQLTimespan.h & https://github.com/ClibMouse/ClickHouse/blob/Kusto-phase3/src/Parsers/Kusto/ParserKQLTimespan.cpp). Otherwise, I think it shouldn't result in a copy just a move.